pkg/metadata/metadata.go

```
@@ -22,6 +23,17 @@ func (m Metadata) ToStructPB() (*structpb.Struct, error) {
 	return structpb.NewStruct(newMap)
 }
 
+func (m Metadata) ToStringValueMap() map[string]string {
+	newMap := make(map[string]string)
+
+	for key, value := range m {
+		val, _ := json.Marshal(value)
```

Let's not ignore the error here. Lets catch it and return the error.